cluster-provision/k8s/check-cluster-up.sh

```diff
@@ -6,85 +6,98 @@ set -exuo pipefail
-  ${ksh} get node node02
-  ${ssh} node01 -- ip l show eth1
-  ${ssh} node01 -- ip l show eth2
-  ${ssh} node02 -- ip l show eth1
-  ${ssh} node02 -- ip l show eth2
-
-  pre_pull_image_file="$DIR/${provision_dir}/extra-pre-pull-images"
-  if [ -f "${pre_pull_image_file}" ]; then
-    bash -x "$DIR/deploy-manifests.sh" "${provision_dir}"
-    bash -x "$DIR/check-pod-images.sh" "${provision_dir}"
-    bash -x "$DIR/validate-pod-pull-policies.sh"
-  fi
-
-  # Run conformance test only at CI and if the provider has them activated
-  conformance_config=$DIR/${provision_dir}/conformance.json
-
-  if [ "${CI}" == "true" -a -f $conformance_config ]; then
-    sonobouy_plugins=""
-
-    if [ "$RUN_KUBEVIRT_CONFORMANCE" == "true" ]; then
-      LATEST=$(curl -L "https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest")
-      ${ksh} apply -f "https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${LATEST}/kubevirt-operator.yaml"
-      ${ksh} apply -f "https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${LATEST}/kubevirt-cr.yaml"
-
-      ${ksh} wait -n kubevirt kv kubevirt --for condition=Available --timeout 15m
-
-      export SONOBUOY_EXTRA_ARGS="--plugin https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${LATEST}/conformance.yaml"
-      hack/conformance.sh $conformance_config
+    ksh="./cluster-up/kubectl.sh"
+    ssh="./cluster-up/ssh.sh"
+    cd "$DIR" && cd ../..
+    export KUBEVIRTCI_PROVISION_CHECK=1
+    export KUBEVIRT_PROVIDER="k8s-${provider}"
+    export KUBEVIRT_NUM_NODES=2
+    export KUBEVIRT_NUM_SECONDARY_NICS=2
+    export KUBEVIRT_WITH_CNAO=true
+    export KUBEVIRT_DEPLOY_ISTIO=true
+    if [[ $KUBEVIRT_PROVIDER =~ k8s-1\.1.* ]]; then
+        export KUBEVIRT_DEPLOY_ISTIO=false
     fi
+    export KUBEVIRT_DEPLOY_PROMETHEUS=true
+    export KUBEVIRT_DEPLOY_PROMETHEUS_ALERTMANAGER=true
+    export KUBEVIRT_DEPLOY_GRAFANA=true
+    if [[ ($KUBEVIRT_PROVIDER =~ k8s-1\.1.*) || ($KUBEVIRT_PROVIDER =~ k8s-1.20) ]]; then
+        export KUBEVIRT_DEPLOY_PROMETHEUS=false
+        export KUBEVIRT_DEPLOY_PROMETHEUS_ALERTMANAGER=false
+        export KUBEVIRT_DEPLOY_GRAFANA=false
+    fi
+    export KUBEVIRT_DEPLOY_CDI=true
+    trap cleanup EXIT ERR SIGINT SIGTERM SIGQUIT
+    bash -x ./cluster-up/up.sh
+
+    # whereabouts ip-reconciler cleanup: ip-reconciler cronjob pods must get removed directly after creation
+    ${ksh} patch cronjob ip-reconciler --type merge -p '{ "spec": { "jobTemplate": { "spec": { "ttlSecondsAfterFinished": 0 } } } }'
 
-    export SONOBUOY_EXTRA_ARGS="--plugin systemd-logs --plugin e2e"
-    hack/conformance.sh $conformance_config
-  fi
+    # whereabouts ip-reconciler cleanup: delete eventually lingering pods before and during the loop
+    delete_ip_reconciler_pods
+    until ${ksh} wait --for=condition=Ready pod --timeout=30s --all; do
+        delete_ip_reconciler_pods
```

Comment:
@dhiller Hi. I see that you've decided to abandon your previous approach of using `timeout 210s bash -c "until ${ksh} wait --for=condition=Ready pod --timeout=30s --all; do delete_ip_reconciler_pods; sleep 1; done"` , wondering if it's because it may take more than 210 secs for the ip-reconciler pods to get cleaned?

Reply:
I did this as the inner bash scope would not know of the declared function. I thought about doing some kind of retry logic, but then thought that the test pod would get killed anyway. WDYT?

Reply:
@dhiller I see what you mean about the inner bash scope. I have another concern, are we sure that at this stage the ip-reconciler pods will always be in `Completed` state? Is the assumption that if they were still executing the actual job we won't get to here?

Reply:
@dhiller FYI you can export the function so it could be used from another subshell `export -f delete_ip_reconciler_pods`